webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/InterpolatedValueCompiler.scala

```diff
@@ -406,64 +405,88 @@ class InterpolatedValueCompilerImpl extends RegexParsers with InterpolatedValueC
     }
   }
 
+}
 
-  //// parsing language
+object PropertyParser {
 
-  //make a case insensitive regex from the parameter
-  //modifier: see http://docs.oracle.com/javase/7/docs/api/java/util/regex/Pattern.html
-  //i: ignore case
-  //u: considere unicode char for ignore case
-  //s: match end of line in ".*" pattern
-  //m: multi-lines mode
-  //exactly quote s - no regex authorized in
-  def id(s: String) = ("""(?iu)\Q""" + s + """\E""").r
+  import fastparse._, NoWhitespace._
 
-  val all       : Parser[List[Token]] = (rep1(interpol   | plainString ) | emptyVar)
+  def parse(value: String): PureResult[List[Token]] = {
+    fastparse.parse(value, all(_)) match {
+      case Parsed.Success(value, index)    => Right(value)
+      case Parsed.Failure(label, i, extra) => Left(Unexpected(s"""Error when parsing value "${value}", error message is: ${label}"""))
+    }
+  }
 
-  val space = """\s*""".r
+  def all[_: P] : P[List[Token]] = P( Start ~ ( (plainStringNoDollar | varOrString).rep(1) | emptyVar) ~ End).map(_.toList)
 
   //empty string is a special case that must be look appart from plain string.
   //also parse full blank string, because no need to look for more case for them.
-  val emptyVar    : Parser[List[CharSeq]] = """(?iums)(\s)*""".r ^^ { x => List(CharSeq(x)) }
-
+  def emptyVar[_: P]: P[List[CharSeq]] = P(CharsWhile(_.isSpaceChar, 0).!).map(x => List(CharSeq(x)) )
```

Comment:
you take only space character here, 

You could write 

s.!.map(x => List(CharSeq(x)))

Reply:
Also I think it is already treated by plainStringNoDollar

Reply:
no, it's needed, I added a test case for it

Reply:
i removed it and added a test case and all tests are ok

Reply:
the problem is with empty string `""`